app/frontend/src/app/IModelBrowser/LocalIModelBrowser.tsx

```diff
@@ -21,14 +21,15 @@ export interface LocalIModelBrowserProps {
 
 export function LocalIModelBrowser(props: LocalIModelBrowserProps): React.ReactElement {
   const backendApi = useBackendApi(props.backendApiPromise);
-  const availableIModels = useAvailableIModels(backendApi);
-  const { displayMode } = React.useContext(iModelBrowserContext);
-
+  const { displayMode, searchQuery } = React.useContext(iModelBrowserContext);
+  const availableIModels = useAvailableIModels(backendApi, searchQuery);
   if (availableIModels?.length === 0) {
     return (
       <VerticalStack className="imodel-browser-no-data">
         <SvgImodelHollow />
-        <Title isMuted>No local iModel snapshots found</Title>
+        <Title isMuted>
+          {searchQuery.length > 0 ? "No local iModels match search query" : "No local iModel snapshots found"}
```

Comment:
```suggestion
          {searchQuery ? "No local iModels match search query" : "No local iModel snapshots found"}
```

In `ITwinBrowser` condition is checking if `searchQuery` is truthy instead of it's length. I would suggest to be consistent.

Reply:
Agree, your suggestion is better because it's shorter.